### PR TITLE
Deprecate `@list` decorator in favor of `rest`'s `@listsResource`

### LINKED ISSUE #1978: Do we need the `@list` decorator in compiler
We have this decorator with no signature and documentation
https://github.com/microsoft/typespec/blob/01b75f905de93c844a8a01449d9b5bb062ebeeec/packages/compiler/lib/decorators.ts#L776

### PR DESCRIPTION
**Pull request by daviwil**
This change fixes #1978 by deprecating the virtually-unused `@list` decorator in favor of the `@listsResource` decorator in `@typespec/rest`.  I'm also deprecating the associated `isListOperation` and `getListOperationType` functions in `@typespec/compiler` and moving `isListOperation` to the REST library.

The corresponding `typespec-azure` PR is here: https://github.com/Azure/typespec-azure/pull/3200

### CODE REVIEW

File: common/changes/@typespec/compiler/move-list-decorator_2023-06-20-11-35.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "changes": [
+    {
+      "packageName": "@typespec/compiler",
+      "comment": "Deprecate `@list` decorator in favor of `@listsResource` in `@typespec/rest`",
```
Comment (timotheeguerin) [inline common/changes/@typespec/compiler/move-list-decorator_2023-06-20-11-35.json:5]:
```suggestion
      "comment": "**Deprecate** `@list` decorator in favor of `@listsResource` in `@typespec/rest`",
```

File: common/changes/@typespec/compiler/move-list-decorator_2023-06-20-11-35.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      "type": "none"
+    },
+    {
+      "packageName": "@typespec/compiler",
+      "comment": "Deprecate `isListOperation` function in favor of `isListOperation` in `@typespec/rest`",
+      "type": "none"
+    },
+    {
+      "packageName": "@typespec/compiler",
+      "comment": "Deprecate `getListOperationType` function",
```
Comment (timotheeguerin) [inline common/changes/@typespec/compiler/move-list-decorator_2023-06-20-11-35.json:15]:
```suggestion
      "comment": "**Deprecate** `getListOperationType` function",
```
just so it is more visible in the changelog

File: docs/standard-library/built-in-decorators.md [hunk 3]
```diff
@@ -305,6 +305,25 @@ Invalid,
 ```
 
 
+### `@list` {#@list}
```
Comment (timotheeguerin) [inline docs/standard-library/built-in-decorators.md:308]: that is unfortunate that we get the doc now that it is deprecated

File: packages/compiler/src/lib/decorators.ts [hunk 4]
```diff
@@ -774,6 +774,9 @@ export function $withoutDefaultValues(context: DecoratorContext, target: Model)
 
 const listPropertiesKey = createStateSymbol("listProperties");
 
+/**
+ * @deprecated Use the `listsResource` decorator in `@typespec/rest` instead.
```
Comment (timotheeguerin) [inline packages/compiler/src/lib/decorators.ts:778]: Would add those entries to https://github.com/microsoft/typespec/issues/1873

File: docs/standard-library/built-in-decorators.md (hunk 3 above)
Comment (timotheeguerin) [reply docs/standard-library/built-in-decorators.md:308]: yeah I guess that is dependent on the design meeting that was inconclusive last week :(

File: packages/compiler/src/lib/decorators.ts (hunk 4 above)
Comment (daviwil) [reply packages/compiler/src/lib/decorators.ts:778]: Will do

Comment (daviwil) [reply packages/compiler/src/lib/decorators.ts:778]: Done

File: packages/rest/test/routes.test.ts [hunk 5]
```diff
@@ -118,7 +118,7 @@ describe("rest: routes", () => {
         @autoRoute
```
[the remark this answers is not in the document]
Comment (daviwil) [reply packages/rest/test/routes.test.ts:118]: Any thoughts on this?

[the remark this answers is not in the document]
Comment (timotheeguerin) [reply packages/rest/test/routes.test.ts:118]: probably just bring it up in the DPG meeting today at 11

[the remark this answers is not in the document]
Comment (daviwil) [reply packages/rest/test/routes.test.ts:118]: Sounds like folks are OK with this one going in!

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**